spec/syntax.md

```diff
@@ -0,0 +1,598 @@
+     by means of GUI tooling, CAT workbenches etc.
+
+1. The syntax surrounding translatable content should be easy to write and edit
+   for developers, localization engineers, and easy to parse by machines.
+
+1. The syntax should make a single message easily embeddable inside many container formats:
+   `.properties`, YAML, XML, inlined as string literals in programming languages, etc.
+   This includes a future _MessageResource_ specification.
+
+1. Simple messages that do not use any placeholders or selectors should (as far as possible)
+   be represented in the syntax with no additional characters than their actual contents.
+
+### Design Restrictions
+
+The syntax specification takes into account the following design restrictions:
+
+1. Whitespace outside the translatable content should be insignificant.
+   It should be possible to define a message entirely on a single line with no ambiguity,
+   as well as to format it over multiple lines for clarity.
+
+1. The syntax should not use nor reserve any keywords in any natural language,
+   such as `if`, `match`, or `let`.
+
+1. The syntax should define as few special characters and sigils as possible.
+
+## Overview & Examples
+
+### Simple Messages
+
+A simple message without any variables does not need any syntax:
+
+    Hello, world!
+
+The same message defined in a `.properties` file:
+
+```properties
+app.greetings.hello = Hello, world!
+```
+
+The same message defined inline in JavaScript:
+
+```js
+let hello = new MessageFormat('Hello, world!')
+hello.format()
+```
+
+### Simple Placeholders
+
+A message with an interpolated variable needs to be interpreted as a pattern,
+which uses `[…]` delimiters:
+
+    [Hello, {$userName}!]
+
+The same message defined in a `.properties` file:
+
+```properties
+app.greetings.hello = [Hello, {$userName}!]
+```
+
+The same message defined inline in JavaScript:
+
+```js
+let hello = new MessageFormat('[Hello, {$userName}!]')
+hello.format({ userName: 'Anne' })
+```
+
+### Formatting Functions
+
+A message with an interpolated `$date` variable formatted with the `datetime` function:
+
+    [Today is {$date: datetime weekday=long}.]
+
+A message with an interpolated `$userName` variable formatted with
+the custom `person` function capable of
+declension (using either a fixed dictionary, algorithmic declension, ML, etc.):
+
+    [Hello, {$userName: person case=vocative}!]
+
+A message with an interpolated `$userObj` variable formatted with
+the custom `person` function capable of
+plucking the first name from the object representing a person:
+
+    [Hello, {$userObj: person firstName=long}!]
+
+### Markup Elements
+
+A message with two markup-like element placeables, `button` and `link`,
+which the runtime can use to construct a document tree structure for a UI framework.
+
+    [{button}Submit{/button} or {link}cancel{/link}.]
+
+### Selection
+
+A message with a single selector:
+
+    {$count: number}
+        one [You have one notification.]
```

Comment:
All examples using `one` are incorrect, should use `1`

They would be OK if the message would contain the `{$count}` to be formatted.

Imagine using `21` instead of `1`.
Showing `"You have 21 notification"` is fine (in Russian :-).
Showing `"You have one notification"` when the count is 21 is incorrect.

Reply:
```suggestion
        1 [You have one notification.]
```